src/main/java/com/flagsmith/FlagsmithClient.java

```diff
@@ -451,33 +472,40 @@ public Builder withFlagsmithApiWrapper(FlagsmithApiWrapper flagsmithApiWrapper)
      * @return a FlagsmithClient
      */
     public FlagsmithClient build() {
-      final FlagsmithApiWrapper flagsmithApiWrapper;
+      if (configuration.getOfflineMode()) {
+        if (configuration.getOfflineHandler() == null) {
+          throw new FlagsmithRuntimeError("Offline handler must be provided to use offline mode.");
+        }
+      }
 
       if (this.flagsmithApiWrapper != null) {
-        flagsmithApiWrapper = this.flagsmithApiWrapper;
+        client.flagsmithSdk = this.flagsmithApiWrapper;
       } else if (cacheConfig != null) {
-        flagsmithApiWrapper = new FlagsmithApiWrapper(
+        client.flagsmithSdk = new FlagsmithApiWrapper(
             cacheConfig.getCache(),
             this.configuration,
             this.customHeaders,
             client.logger,
             apiKey);
       } else {
-        flagsmithApiWrapper = new FlagsmithApiWrapper(
+        client.flagsmithSdk = new FlagsmithApiWrapper(
             this.configuration,
             this.customHeaders,
             client.logger,
             apiKey);
       }
 
-      client.flagsmithSdk = flagsmithApiWrapper;
-
       if (configuration.getAnalyticsProcessor() != null) {
-        configuration.getAnalyticsProcessor().setApi(flagsmithApiWrapper);
+        configuration.getAnalyticsProcessor().setApi(client.flagsmithSdk);
         configuration.getAnalyticsProcessor().setLogger(client.logger);
       }
 
       if (configuration.getEnableLocalEvaluation()) {
+        if (configuration.getOfflineHandler() != null) {
+          throw new FlagsmithRuntimeError(
+              "Local evaluation and offline handler cannot be used together.");
+        }
+
```

Comment:
I wonder if it's more correct to have this validation logic in the `FlagsmithConfig` constructor?

Reply:
I guess my commend contradicts https://github.com/Flagsmith/flagsmith-java-client/issues/73 which I don't fully understand yet.

Reply:
Yeah, I considered this but the whole relationship between FlagsmithConfig and FlagsmithClient is a bit of a mess right now so I didn't spend too much time on it.